#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/gctronic/e-puck/protos/E-puck.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.4878394251964574 -0.46536454959373375 0.738544874198861 4.369098742076773
  position -0.22848916649411444 0.1962209290334096 1.0457874888712706
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorAppearance Appearance {
    material Material {
    }
    texture ImageTexture {
      url [
        "../elipse.png"
      ]
      repeatS FALSE
      repeatT FALSE
    }
    textureTransform TextureTransform {
      scale 0.5 0.5
    }
  }
}
E-puck {
  hidden position_0_0 1656.949931112391
  hidden position_0_1 1606.9573561617415
  hidden linearVelocity_0 1.3373042858274196e-11 -1.3892957895222936e-11 2.7374369437893623e-12
  hidden angularVelocity_0 -5.551402907739319e-11 -1.3532555050732762e-10 -5.1125203184913994e-08
  hidden rotation_1 0 -0.9999999999999999 0 1.810989983019916
  hidden linearVelocity_1 -2.6204496350558914e-10 -1.3133611531649986e-09 6.22599749494672e-12
  hidden angularVelocity_1 0.9780820748399524 -0.20821603884226508 -0.0010786699098995693
  hidden rotation_2 0 -1 0 1.5380824762326657
  hidden linearVelocity_2 2.896882733699352e-10 1.2850939867814658e-09 -1.1212364370283289e-12
  hidden angularVelocity_2 0.9780820919739746 -0.2082159600622576 -0.0010786899272980587
  translation -0.4631765093694154 -0.4632357774251278 -3.6099195311707286e-05
  rotation -0.002798471959163859 -0.0013726990084096998 -0.9999951421142634 1.7805493231070237
  controller "<none>"
  groundSensorsSlot [
    DistanceSensor {
      translation 0 0.03 0
      rotation 0 1 0 1.5708
      name "ir2left"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 0.02 0
      rotation 0 1 0 1.5708
      name "ir1left"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 0.01 0
      rotation 0 1 0 1.5708
      name "ir0left"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 -0.01 0
      rotation 0 1 0 1.5708
      name "ir0right"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 -0.03 0
      rotation 0 1 0 1.5708
      name "ir2right"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 -0.02 0
      rotation 0 1 0 1.5708
      name "ir1right"
      type "infra-red"
    }
  ]
}
Robot {
  translation -0.20937420236442864 -0.24427900453532184 0.012641679585965565
  rotation 0.0337003442022856 -0.005563246336841158 -0.9994164983083095 -2.813376248638155
  children [
    DEF Sensors Group {
      children [
        DistanceSensor {
          translation 0.02 0.01 -0.0053998
          rotation 0 1 0 1.5708
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 1 0 0
                metalness 0
              }
              geometry Box {
                size 0.0005 0.003 0.001
              }
            }
          ]
          name "ir0left"
          type "infra-red"
        }
        DistanceSensor {
          translation 0.02 0.02 -0.0053998
          rotation 0 1 0 1.5708
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 1 0 0
                metalness 0
              }
              geometry Box {
                size 0.0005 0.003 0.001
              }
            }
          ]
          name "ir1left"
          type "infra-red"
        }
        DistanceSensor {
          translation 0.02 0.03 -0.0053998
          rotation 0 1 0 1.5708
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 1 0 0
                metalness 0
              }
              geometry Box {
                size 0.0005 0.003 0.001
              }
            }
          ]
          name "ir2left"
          type "infra-red"
        }
        DistanceSensor {
          translation 0.02 -0.01 -0.0053998
          rotation 0 1 0 1.5708
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 1 0 0
                metalness 0
              }
              geometry Box {
                size 0.0005 0.003 0.001
              }
            }
          ]
          name "ir0right"
          type "infra-red"
        }
        DistanceSensor {
          translation 0.02 -0.02 -0.0053998
          rotation 0 1 0 1.5708
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 1 0 0
                metalness 0
              }
              geometry Box {
                size 0.0005 0.003 0.001
              }
            }
          ]
          name "ir1right"
          type "infra-red"
        }
        DistanceSensor {
          translation 0.02 -0.03 -0.0053998
          rotation 0 1 0 1.5708
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 1 0 0
                metalness 0
              }
              geometry Box {
                size 0.0005 0.003 0.001
              }
            }
          ]
          name "ir2right"
          type "infra-red"
        }
      ]
    }
    DEF right_wheel HingeJoint {
      jointParameters HingeJointParameters {
        position 2035.2473410538091
        axis 0 1 0
        anchor -0.02 -0.031 -1.01962e-08
      }
      device [
        RotationalMotor {
          name "right wheel motor"
        }
      ]
      endPoint Solid {
        translation -0.01999917063578704 -0.0305 3.841210945098771e-07
        rotation 0.9394847677297009 0.24227208108418014 0.2422242967396341 4.650054878896689
        children [
          DEF wheel1 Shape {
            appearance PBRAppearance {
              baseColor 0 0 0
              metalness 0
            }
            geometry Cylinder {
              height 0.011
              radius 0.014
            }
          }
        ]
        name "solid(1)"
        boundingObject USE wheel1
        physics Physics {
        }
        linearVelocity -0.012099251814061495 0.0041140818174405635 5.8697933653679885e-08
        angularVelocity -0.29811544138070534 -0.8767862273230016 -0.03392947700199833
      }
    }
    DEF left_wheel HingeJoint {
      jointParameters HingeJointParameters {
        position 2360.308088074732
        axis 0 1 0
        anchor -0.02 0.031 -1.01962e-08
      }
      device [
        RotationalMotor {
          name "left wheel motor"
        }
      ]
      endPoint Solid {
        translation -0.01999968015277221 0.0305 2.6020909810111608e-06
        rotation 0.3498664771506979 0.6623512092249535 0.6624834517211747 3.814936023539664
        children [
          DEF wheel2 Shape {
            appearance PBRAppearance {
              baseColor 0 0 0
              metalness 0
            }
            geometry Cylinder {
              height 0.011
              radius 0.014
            }
          }
        ]
        name "solid(2)"
        boundingObject USE wheel2
        physics Physics {
        }
        linearVelocity -0.014054502720232289 0.00478944441673104 -5.882099036990118e-08
        angularVelocity -0.3463942176926954 -1.0165662819463535 -0.03402173952959134
      }
    }
    DEF BODY Solid {
      children [
        DEF body Group {
          children [
            Pose {
              translation -0.02 0 0
              children [
                Shape {
                  appearance PBRAppearance {
                  }
                  geometry Box {
                    size 0.02 0.05 0.01
                  }
                }
              ]
            }
            Pose {
              children [
                Shape {
                  appearance PBRAppearance {
                  }
                  geometry Box {
                    size 0.02 0.03 0.01
                  }
                }
              ]
            }
            Pose {
              translation 0.015 0 0
              children [
                Shape {
                  appearance PBRAppearance {
                  }
                  geometry Box {
                    size 0.02 0.01 0.01
                  }
                }
              ]
            }
            Pose {
              translation 0.02 2.03924e-08 -0.004
              rotation 0 0 1 -1.5707953071795862
              children [
                Shape {
                  appearance PBRAppearance {
                  }
                  geometry Box {
                    size 0.075 0.01 0.002
                  }
                }
              ]
            }
            Pose {
              translation 0.01 0 -0.008
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColor 0 0 0
                    metalness 0
                  }
                  geometry Sphere {
                    radius 0.004
                    subdivision 2
                  }
                }
              ]
            }
          ]
        }
      ]
    }
  ]
  boundingObject USE body
  physics Physics {
  }
  controller "line_follower"
  linearVelocity -0.012931639664202974 0.004867284388515125 3.5848418055991206e-07
  angularVelocity -6.448200773333228e-07 -2.2758053298991293e-05 -0.033868307478580946
}
